lib/ansible/modules/copy.py

```diff
@@ -680,6 +683,11 @@ def main():
                         module.set_owner_if_different(src, owner, False)
                     if group is not None:
                         module.set_group_if_different(src, group, False)
+
+                    secontext = file_args['secontext']
+
+                    if secontext and any(secontext):  # only bother if any of the context attributes were specified
+                        module.set_context_if_different(src, secontext, False)
```

Comment:
in cases the tempfile is not in the same folder, this might fail, so i would make this 'best effort'